sycl/test-e2e/ESIMD/rdtsc.cpp

```diff
@@ -0,0 +1,78 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+//==- rdtsc.cpp - Test to verify rdtsc0 and sr0 functionlity----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// This is basic test to validate rdtsc function.
+
+#include <cmath>
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/ext/intel/esimd/simd.hpp>
+#include <sycl/sycl.hpp>
+#include <vector>
+
+int ErrCnt = 0;
+template <typename DataT>
+using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
+template <typename DataT>
+using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
+
+int test_rdtsc_sr0() {
+  sycl::queue Queue;
+  shared_allocator<uint64_t> Allocator(Queue);
+  constexpr int32_t SIZE = 32;
+
+  shared_vector<uint64_t> VectorOutputRDTSC(SIZE, -1, Allocator);
+
+  auto GlobalRange = sycl::range<1>(SIZE);
+  sycl::range<1> LocalRange{1};
+  sycl::nd_range<1> Range(GlobalRange, LocalRange);
+
+  {
+    Queue.submit([&](sycl::handler &cgh) {
+      uint64_t *VectorOutputRDTSCPtr = VectorOutputRDTSC.data();
+
+      auto Kernel = ([=](sycl::nd_item<1> ndi) [[intel::sycl_explicit_simd]] {
+        using namespace sycl::ext::intel::esimd;
+        auto Idx = ndi.get_global_id(0);
+        simd<uint64_t, SIZE> DummyVector;
+        uint64_t StartCounter = sycl::ext::intel::experimental::esimd::rdtsc();
+        DummyVector.copy_from(VectorOutputRDTSCPtr);
+        uint64_t EndCounter = sycl::ext::intel::experimental::esimd::rdtsc();
+
+        simd<uint64_t, 1> VectorResultRDTSC = EndCounter - StartCounter;
```

Comment:
This is too easy for DCE to eliminate DummyVector completely.
```suggestion
        simd<uint64_t, SIZE> DummyVector;
        uint64_t StartCounter = sycl::ext::intel::experimental::esimd::rdtsc();
        simd<uint64_t, 1> VectorResultRDTSC(VectorOutputRDTSCPtr + Idx);
        DummyVector.copy_from(VectorOutputRDTSCPtr);
        uint64_t EndCounter = sycl::ext::intel::experimental::esimd::rdtsc();

        VectorResultRDTSC += EndCounter - StartCounter;
```